src/lib.rs

```diff
@@ -410,14 +410,19 @@ fn process_join(from: &Arc<Session>, room_id: RoomId, user_id: UserId, subscribe
             match ValidatedToken::from_str(token, key) {
                 Ok(tok) => {
                     janus_info!("Processing validated join from {:p} to room ID {} with user ID {}. Join allowed: {}", from.handle, room_id, user_id, tok.join_hub);
+                    if !tok.join_hub {
+                        return Err(From::from("Rejecting join with no join_hub permission!"))
+                    }
```

Comment:
We should probably eliminate this log message in the expected case, but it's low-volume anyway so it doesn't matter much.